packages/editor/src/core/helpers/use-shadow-root.ts

```diff
@@ -22,3 +22,19 @@ export function useShadowRoot(containerRef: React.RefObject<HTMLElement>) {
 
   return shadowRoot
 }
+
+export function isShadowRoot(
+  shadowRoot: ShadowRoot | null | Node | undefined
+): shadowRoot is ShadowRoot {
+  return shadowRoot instanceof ShadowRoot
+}
+
+export function getFirstElementOrUndefined(
+  shadowRoot: ShadowRoot | null | undefined
+): HTMLElement | undefined {
+  if (!shadowRoot) {
+    return undefined
+  }
+
+  return shadowRoot?.firstElementChild as HTMLElement
+}
```

Comment:
possible simplifications:

```suggestion
export function getFirstElementOrUndefined(shadowRoot?: ShadowRoot | null) {
  return shadowRoot?.firstElementChild as HTMLElement | undefined
}
```

Reply:
I like the way I wrote it better. Some more lines, but more clarity and easier types